src/olympia/bandwagon/forms.py

```diff
@@ -195,43 +196,49 @@ def clean_icon(self):
         icon = self.cleaned_data['icon']
         if not icon:
             return
-        if icon.content_type not in ('image/png', 'image/jpeg'):
+        icon_check = ImageCheck(icon)
+        if (icon.content_type not in ('image/png', 'image/jpeg') or
+                not icon_check.is_image()):
             raise forms.ValidationError(
                 ugettext('Icons must be either PNG or JPG.'))
 
+        if icon_check.is_animated():
+            raise forms.ValidationError(ugettext('Icons cannot be animated.'))
+
         if icon.size > settings.MAX_ICON_UPLOAD_SIZE:
             size_in_mb = settings.MAX_ICON_UPLOAD_SIZE / 1024 / 1024 - 1
             raise forms.ValidationError(
                 ugettext('Please use images smaller than %dMB.') % size_in_mb)
         return icon
 
     def save(self, default_locale=None):
-        c = super(CollectionForm, self).save(commit=False)
-        c.author = self.initial['author']
-        c.application = self.initial['application']
+        collection = super(CollectionForm, self).save(commit=False)
+        collection.author = self.initial['author']
+        collection.application = self.initial['application']
         icon = self.cleaned_data.get('icon')
 
         if default_locale:
-            c.default_locale = default_locale
+            collection.default_locale = default_locale
 
         if icon:
-            c.icontype = 'image/png'
+            collection.icontype = 'image/png'
 
-        c.save()
+        collection.save()
 
         if icon:
-            dirname = c.get_img_dir()
+            dirname = collection.get_img_dir()
 
-            destination = os.path.join(dirname, '%d.png' % c.id)
-            tmp_destination = os.path.join(dirname,
-                                           '%d.png__unconverted' % c.id)
+            destination = os.path.join(dirname, '%d.png' % collection.id)
+            tmp_destination = os.path.join(
+                dirname, '%d.png__unconverted' % collection.id)
+            icon.seek(0)
```

Comment:
Am I blind or where is the `icon` read before this? Or is this primarily "safety first" just in case it got read by another method?

Reply:
We're in `save`, the `ImageCheck` in `clean_icon()` would have read it.

Reply:
gotcha, that makes sense. Maybe add a comment?